File: apps/backend-java/src/main/java/com/housingplatform/features/properties/dto/PropertyDetail.java
package com.housingplatform.features.properties.dto;

import com.housingplatform.persistence.enums.AccommodationType;
import com.housingplatform.persistence.enums.BookingMode;
import java.util.List;
import java.util.UUID;

public record PropertyDetail(
    UUID id,
    String title,
    String slug,
    String description,
    AccommodationType propertyType,
    String district,
    String nearestStationName,
    Integer nearestStationWalkMin,
    String addressLine1,
    String addressLine2,
    String city,
    BookingMode bookingMode,
    int minStayNights,
    int monthlyPriceMin,
    List<String> tags,
    String hostDisplayName,
    Double latitude,
    Double longitude,
    List<PropertyImageDto> images,
    List<PropertyRoomDto> rooms,
    List<PropertyAmenityDto> amenities) {}

File: apps/backend-node/src/features/properties/dto/property-detail.ts
import type { AccommodationType } from './accommodation-type';
import type { BookingMode } from './booking-mode';
import type { PropertyAmenityDto } from './property-amenity';
import type { PropertyImageDto } from './property-image';
import type { PropertyRoomDto } from './property-room';

export interface PropertyDetail {
  id: string;
  title: string;
  slug: string;
  description: string;
  propertyType: AccommodationType;
  district: string;
  nearestStationName: string | null;
  nearestStationWalkMin: number | null;
  addressLine1: string;
  addressLine2: string | null;
  city: string;
  bookingMode: BookingMode;
  minStayNights: number;
  monthlyPriceMin: number;
  tags: string[];
  hostDisplayName: string;
  latitude: number | null;
  longitude: number | null;
  images: PropertyImageDto[];
  rooms: PropertyRoomDto[];
  amenities: PropertyAmenityDto[];
}
